src/label/labelStyle.ts

```diff
@@ -292,6 +293,66 @@ function setLabelStyle<TLabelDataIndex>(
 }
 export { setLabelStyle };
 
+/**
+ * Set and create specific labels for box plot.
+ *
+ * This method should only be used for this case,
+ * as it manually creates multiple alternating labels
+ * specifically for the box plot.
+ */
+export function setBoxLabels(boxItemPositions: number[][], formattedLabels: Array<string>,
+    labelOption: any, group: ViewRootGroup) {
+    // get sorted and unique y positions of box items
+    const yPositions = boxItemPositions.map((pos: number[]) => pos[1]);
+    const uniqueYPositions: number[] = [];
+    yPositions.forEach(position => {
+        if (!uniqueYPositions.includes(position)) {
+            uniqueYPositions.push(position);
+        }
+    });
+    uniqueYPositions.sort(function (a: number, b: number) {
+        return b - a;
+    });
+
+    boxItemPositions.sort(function (a: number[], b: number[]) {
+        return a[0] - b[0];
+    });
+
+    // get alternating y positions for labels to avoid overlap
+    const uniqueAlternatingPositions = uniqueYPositions.map((posY: number, ind: number) => {
+            const matchingPositions = boxItemPositions.filter((orgPos: number[]) => orgPos[1] === posY);
+            const index = (ind % 2 === 0) ? 0 : matchingPositions.length - 1;
+            return matchingPositions[index];
+    });
+
+    // create labels and add them to their respective positions
+    formattedLabels.forEach((labelText: string, ind: number) => {
+        if (labelOption.show === 'iqr' && (ind === 0 || ind === 4)) {
+            return;
+        }
+        if (labelOption.show === 'median' && (ind !== 2)) {
+            return;
+        }
+        if (labelOption.show === 'whiskers' && (ind !== 0 && ind !== 4)) {
+            return;
+        }
+        const label = new graphic.Text({
+            style: {
+                text: labelText
+            },
+            z2: 1000
+        });
+        const defaultOffset = 5;
+        const defaultXOffset = (ind % 2) === 0 ? (-(label.getBoundingRect().width + defaultOffset)) : defaultOffset;
```

Comment:
As I said before, we should provide a way to let developers decide where to display the label to the left or right. The `label.align` can be changed into `'left' | 'right' | 'center' | { iqr: 'left' | 'right' | 'center' , ... }`.

Reply:
@Ovilia Sorry, maybe I am missing something here as well but there is no single label. We have five labels, right? I still don't get how someone could choose the align for the five in a way that makes sense - e.g. all labels on one side - even though they might overlap?